pkgs/by-name/at/atlauncher/package.nix

```diff
@@ -20,27 +19,38 @@
   xorg,
 }:
 
-stdenvNoCC.mkDerivation (finalAttrs: {
+let
   pname = "atlauncher";
   version = "3.4.36.10";
 
-  src = fetchurl {
-    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${finalAttrs.version}/ATLauncher-${finalAttrs.version}.jar";
-    hash = "sha256-JZTiYcea5ik8a4RmNLxZcuea7spGWftUGRiRW2Ive7c=";
+  src = fetchFromGitHub {
+    owner = "ATLauncher";
+    repo = "ATLauncher";
+    rev = "v${version}";
+    sha256 = "sha256-EMV/M5W/acJaQJsi4COoRQC2YVfuJG9HGcwvGfakoNo=";
   };
-
-  env.ICON = fetchurl {
-    url = "https://atlauncher.com/assets/images/logo.svg";
-    hash = "sha256-XoqpsgLmkpa2SdjZvPkgg6BUJulIBIeu6mBsJJCixfo=";
-  };
-
-  dontUnpack = true;
+in
+stdenvNoCC.mkDerivation {
+  inherit pname version src;
 
   nativeBuildInputs = [
-    copyDesktopItems
+    gradle
     makeWrapper
   ];
 
+  mitmCache = gradle.fetchDeps {
+    inherit pname;
+    data = ./deps.json;
+  };
+
+  gradleFlags = [
+    "build"
```

Comment:
```suggestion
  doCheck = true;

  gradleFlags = [
```

Reply:
you didn't apply this

Reply:
What does `doCheck = true` do under-the-hood? Since I'm explicitly excluding the test task, or am I understanding this incorrectly?

Reply:
it simply runs the `gradleCheckTask` (defaults to `test`) in the check phase.